File: Android project/AndroidNetworkniginJava/app/src/main/java/com/mirkamol/androidnetworkniginjava/network/volley/MyApplication.java
package com.mirkamol.androidnetworkniginjava.network.volley;

import static android.content.ContentValues.TAG;

import android.app.Application;


import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class MyApplication extends Application {
    public static MyApplication instance;
    private RequestQueue requestQueue;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
    public RequestQueue getRequestQueue(){
        if (requestQueue == null)
            requestQueue = Volley.newRequestQueue(this);
        return requestQueue;
    }

    public <T> void addToRequestQueue(Request<T> request){
        request.setTag(TAG);
        getRequestQueue().add(request);
    }

}

File: Android project/Lesson6VolleyNetWorking/app/src/main/java/com/mirkamol/lesson6volleynetworking/activities/MyApplication.kt
package com.mirkamol.lesson6volleynetworking.activities

import android.app.Application
import android.util.Log
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import com.mirkamol.lesson6volleynetworking.network.VolleyHttp.Companion.TAG

class MyApplication:Application() {

    companion object{
        var instance:MyApplication? = null
    }


    override fun onCreate() {
        super.onCreate()
        instance = this
        Log.d(TAG, "onCreate: ")
    }

    private val requestQueue: RequestQueue? = null
        get() {
            if (field == null) {
                return Volley.newRequestQueue(applicationContext)
            }
            return field
        }

    fun <T> addToRequestQueue(request: Request<T>) {
        request.tag = TAG
        requestQueue?.add(request)
    }
}